public/app/features/dashboard/history/history.html

```diff
@@ -0,0 +1,161 @@
+											<i class="fa fa-rotate-right"></i>&nbsp;&nbsp;Restore
+										</a>
+										<a class="btn btn-outline-disabled btn-small" ng-show="revision.version === ctrl.dashboard.version">
+											<i class="fa fa-check"></i>&nbsp;&nbsp;Current
+										</a>
+									</td>
+								</tr>
+							</tbody>
+						</table>
+
+					<div ng-if="ctrl.appending">
+						<i class="fa fa-spinner fa-spin"></i>
+						<em>Fetching more entries&hellip;</em>
+					</div>
+
+						<div class="gf-form-group" ng-show="ctrl.mode === 'list'">
+							<div class="gf-form-button-row">
+								<a	type="button"
+										class="btn gf-form-button btn-primary"
+										ng-if="ctrl.revisions.length > 1"
+										ng-class="{disabled: !ctrl.isComparable()}"
+										ng-click="ctrl.getDiff(ctrl.diff)"
+										bs-tooltip="ctrl.isComparable() ? '' : 'Select 2 versions to start comparing'">
+									<i class="fa fa-code-fork" ></i>&nbsp;&nbsp;Compare versions
+								</a>
+								<a  type="button"
+										class="btn gf-form-button btn-inverse"
+										ng-if="ctrl.revisions.length >= ctrl.limit"
+										ng-click="ctrl.addToLog()"
+										ng-class="{disabled: ctrl.isLastPage()}"
+										ng-disabled="ctrl.isLastPage()">
+									Show more versions
+								</a>
+							</div>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+
+		<div class="history-list" ng-if="ctrl.mode === 'compare'">
+			<div class="page-container">
+				<div class="page-body">
+					<aside class="page-sidebar">
+						<section class="page-sidebar-section">
+							<ul class="ui-list">
+								<li><a ng-class="{active: ctrl.diff === 'basic'}" ng-click="ctrl.getDiff('basic')" href="">Change Summary</a></li>
+								<li><a ng-class="{active: ctrl.diff === 'html'}" ng-click="ctrl.getDiff('html')" href="">JSON Code View</a></li>
+							</ul>
+						</section>
+					</aside>
+
+					<div class="tab-content page-content-with-sidebar">
+						<div ng-if="ctrl.loading">
+							<i class="fa fa-spinner fa-spin"></i>
+							<em>Fetching changes&hellip;</em>
+						</div>
+
+						<div ng-if="!ctrl.loading" ng-init="new = ctrl.selected[0]; original = ctrl.selected[1]">
+							<a type="button"
+								class="btn gf-form-button btn-primary diff-restore-btn"
+								ng-click="ctrl.restore(new)"
+								ng-if="ctrl.isOriginalCurrent()">
+								<i class="fa fa-rotate-right" ></i>&nbsp;&nbsp;Restore to version {{new}}
+							</a>
+							<h4>
+								Comparing Version {{ctrl.selected[0]}}
+								<i class="fa fa-arrows-h"></i>
+								Version {{ctrl.selected[1]}}
+								<cite class="muted" ng-if="ctrl.isOriginalCurrent()">(Current)</cite>
+							</h4>
+							<section ng-if="ctrl.diff === 'basic'">
+								<p class="small muted">
+									<strong>Version {{new}}</strong> updated by
+									<span>{{ctrl.getMeta(new, 'createdBy')}} </span>
```

Comment:
for future reference, never use functions in angular bindings, bad performance & bad practice. Create view models instead.